contracts/GroupNFT.sol

```diff
@@ -0,0 +1,142 @@
+        }
+        __ERC721_init("Programmable IP Group IP NFT", "GroupNFT");
+        __AccessManaged_init(accessManager);
+        __UUPSUpgradeable_init();
+        _getGroupNFTStorage().imageUrl = imageUrl;
+    }
+
+    /// @dev Sets the Licensing Image URL.
+    /// @dev Enforced to be only callable by the protocol admin
+    /// @param url The URL of the Licensing Image
+    function setLicensingImageUrl(string calldata url) external restricted {
+        GroupNFTStorage storage $ = _getGroupNFTStorage();
+        $.imageUrl = url;
+        emit BatchMetadataUpdate(0, $.totalSupply);
+    }
+
+    /// @notice Mints a Group NFT.
+    /// @param minter The address of the minter.
+    /// @param receiver The address of the receiver of the minted Group NFT.
+    /// @return groupNftId The ID of the minted Group NFT.
+    function mintGroupNft(address minter, address receiver) external onlyIPAssetRegistry returns (uint256 groupNftId) {
+        GroupNFTStorage storage $ = _getGroupNFTStorage();
+        groupNftId = $.totalSupply++;
+        _mint(receiver, groupNftId);
+        emit GroupNFTMinted(minter, receiver, groupNftId);
+    }
+
+    /// @notice Returns the total number of minted group IPA NFT since beginning,
+    /// @return The total number of minted group IPA NFT.
+    function totalSupply() external view returns (uint256) {
+        return _getGroupNFTStorage().totalSupply;
+    }
+
+    /// @notice ERC721 OpenSea metadata JSON representation of Group IPA NFT
+    function tokenURI(
+        uint256 id
+    ) public view virtual override(ERC721Upgradeable, IERC721Metadata) returns (string memory) {
+        GroupNFTStorage storage $ = _getGroupNFTStorage();
+
+        /* solhint-disable */
+        // Follows the OpenSea standard for JSON metadata
+
+        // base json, open the attributes array
+        string memory json = string(
+            abi.encodePacked(
+                "{",
+                '"name": "Story Protocol IP Assets Group #',
+                id.toString(),
+                '",',
+                '"description": IPAsset Group",',
+                '"external_url": "https://protocol.storyprotocol.xyz/ipa/',
+                id.toString(),
+                '",',
+                '"image": "',
+                $.imageUrl,
+                '"'
+            )
+        );
+
+        // close the attributes array and the json metadata object
+        json = string(abi.encodePacked(json, "}"));
+
+        /* solhint-enable */
+
+        return string(abi.encodePacked("data:application/json;base64,", Base64.encode(bytes(json))));
+    }
+
+    /// @notice IERC165 interface support.
+    function supportsInterface(
+        bytes4 interfaceId
+    ) public view virtual override(ERC721Upgradeable, IERC165) returns (bool) {
+        return interfaceId == type(IGroupNFT).interfaceId || super.supportsInterface(interfaceId);
+    }
+
+    ////////////////////////////////////////////////////////////////////////////
```

Comment:
For consistency with other contracts we could remove this comment banner saying "Upgrades related" for all contracts that have them

Reply:
Removed.